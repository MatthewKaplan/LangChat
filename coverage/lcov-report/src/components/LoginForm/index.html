<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/LoginForm</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/LoginForm
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
<<<<<<< HEAD
        <span class="strong">84.21% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>32/38</span>
=======
        <span class="strong">83.78% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>31/37</span>
>>>>>>> master
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.62% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>11/13</span>
      </div>
      <div class='fl pad1y space-right2'>
<<<<<<< HEAD
        <span class="strong">86.11% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>31/36</span>
=======
        <span class="strong">85.71% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>30/35</span>
>>>>>>> master
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="LoginForm.js"><a href="LoginForm.js.html">LoginForm.js</a></td>
<<<<<<< HEAD
	<td data-value="84.21" class="pic high"><div class="chart"><div class="cover-fill" style="width: 84%;"></div><div class="cover-empty" style="width:16%;"></div></div></td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="38" class="abs high">32/38</td>
=======
	<td data-value="83.78" class="pic high"><div class="chart"><div class="cover-fill" style="width: 83%;"></div><div class="cover-empty" style="width:17%;"></div></div></td>
	<td data-value="83.78" class="pct high">83.78%</td>
	<td data-value="37" class="abs high">31/37</td>
>>>>>>> master
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="84.62" class="pct high">84.62%</td>
	<td data-value="13" class="abs high">11/13</td>
<<<<<<< HEAD
	<td data-value="86.11" class="pct high">86.11%</td>
	<td data-value="36" class="abs high">31/36</td>
=======
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="35" class="abs high">30/35</td>
>>>>>>> master
	</tr>

<tr>
	<td class="file empty" data-value="styles.js"><a href="styles.js.html">styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
<<<<<<< HEAD
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jul 24 2019 23:12:14 GMT-0600 (Mountain Daylight Time)
=======
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jul 24 2019 22:34:25 GMT-0600 (Mountain Daylight Time)
>>>>>>> master
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
