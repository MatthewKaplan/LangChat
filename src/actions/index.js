export const changeEmail = text => ({
	type: 'CHANGE_EMAIL',
	text
});

export const changePassword = text => ({
	type: 'CHANGE_PASSWORD',
	text
});

export const addMessage = message => ({
	type: 'ADD_MESSAGE',
	payload: { message }
});
