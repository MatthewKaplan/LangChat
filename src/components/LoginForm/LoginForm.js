import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { Button, Input } from '../common';
import { Actions } from 'react-native-router-flux';
import { MaterialCommunityIcons, AntDesign, EvilIcons } from '@expo/vector-icons';
import { connect } from 'react-redux';
import { emailChanged, passwordChanged } from '../../actions/index';

class LoginForm extends Component {
  
  handleChange = (text, input) => {
    if( input === "emailChanged" ) {
      this.props.emailChanged(text);
    } else {
      this.props.passwordChanged(text);
    }
  }

	render () {
    const { containerStyle, inputContainerStyle, textHeaderStyle, buttonContainerStyle } = styles;
    console.log(this.props.email)
    console.log(this.props.password)
		return (
			<View style={containerStyle}>
				<EvilIcons name="close" size={40} onPress={() => Actions.splashPage()} style={{ width: '13%' }} />
				<Text style={textHeaderStyle}>Log In</Text>
				<View style={inputContainerStyle}>
					<Input
						label={<MaterialCommunityIcons name="email-outline" size={30} color="#999" />}
            placeholder="Email"
            value={this.props.email}
            onChangeText={email => this.handleChange(email, "emailChanged")}
            required
					/>
				</View>
				<View style={inputContainerStyle}>
					<Input
						label={<AntDesign name="lock" size={30} color="#999" />}
						placeholder="Password"
            secureTextEntry
            value={this.props.password}
						onChangeText={password => this.handleChange(password, "passwordChanged")}
					/>
				</View>
				<View style={buttonContainerStyle}>
					<Button>Log In</Button>
				</View>
			</View>
		);
	}
}

const styles = {
	inputContainerStyle: {
		borderBottomWidth: 1,
		padding: 5,
		backgroundColor: '#fff',
		justifyContent: 'flex-start',
		flexDirection: 'row',
		borderColor: '#ddd',
		position: 'relative'
	},
	containerStyle: {
		borderWidth: 1,
		borderRadius: 2,
		borderColor: '#ddd',
		borderBottomWidth: 0,
		shadowColor: '#000',
		shadowOffset: { width: 0, height: 2 },
		shadowOpacity: 0.1,
		shadowRadius: 2,
		elevation: 1,
		marginLeft: 5,
		marginRight: 5,
		paddingTop: 50
	},
	textHeaderStyle: {
		fontSize: 30,
		fontWeight: '600',
		paddingTop: 30,
		paddingBottom: 40
	},
	buttonContainerStyle: {
		marginTop: 80
	}
};

export const mapStateToProps = state => ({
	email: state.auth.email,
	password: state.auth.password
});

export default connect(mapStateToProps, { emailChanged, passwordChanged })(LoginForm);
