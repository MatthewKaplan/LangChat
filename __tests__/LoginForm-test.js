// __tests__/LoginForm-test.js
import React from 'react';
import { LoginForm, mapDispatchToProps } from '../src/components/LoginForm/LoginForm';
import * as actions from '../src/actions/index';
import { shallow } from 'enzyme';

const mockChangeUserName = jest.fn();
const mockChangePassword = jest.fn();
const mockLogin = jest.fn();
const mockUsername = 'mockUsername';
const mockPassword = 'mockPassword';
const mockSplashPage = jest.fn();
const mockHandleError = jest.fn();

describe('LoginForm', () => {
	let wrapper, instance;

	beforeEach(() => {
		wrapper = shallow(
			<LoginForm
				changeUsername={mockChangeUserName}
				changePassword={mockChangePassword}
				login={mockLogin}
				username={mockUsername}
				password={mockPassword}
				splashPage={mockSplashPage}
				handleError={mockHandleError}
			/>
		);
		instance = wrapper.instance();

		window.fetch = jest.fn().mockImplementation(() =>
			Promise.resolve({
				ok: true,
				json: () => Promise.resolve(1)
			})
		);
	});

	it(`matches snapshot`, () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should match init state', () => {
		const initState = {
			loading: false
		};

		expect(wrapper.state()).toEqual(initState);
	});

	it.skip("should invoke 'Actions.splashPage()' when close button is pressed", () => {
		wrapper.find("[data-test='close-btn']").simulate('press');
		expect(mockSplashPage).toHaveBeenCalled();
	});

	describe('handleChange', () => {
		it("should invoke 'handleChange' when user types username with correct params", () => {
			let mockFn = jest.spyOn(instance, 'handleChange');
			wrapper.find("[data-test='username-input']").simulate('ChangeText', 'mockUser');
			expect(mockFn).toHaveBeenCalledWith('mockUser', 'username');
		});

		it("should invoke 'handleChange' when user types password with correct params", () => {
			let mockFn = jest.spyOn(instance, 'handleChange');
			wrapper.find("[data-test='password-input']").simulate('ChangeText', 'mockPassword');
			expect(mockFn).toHaveBeenCalledWith('mockPassword', 'password');
		});

		it("should invoke 'changeUserName' when handleChange is called with correct params", () => {
			instance.handleChange('mockUser', 'username');
			expect(mockChangeUserName).toHaveBeenCalledWith('mockUser');
		});

		it("should invoke 'changePassword' when handleChange is called with correct params", () => {
			instance.handleChange('mockPassword', 'password');
			expect(mockChangePassword).toHaveBeenCalledWith('mockPassword');
		});
	});

	describe('login', () => {
		it('should set the state of loading to true when login is invoked', () => {
			expect(wrapper.state('loading')).toEqual(false);
			instance.login();
			expect(wrapper.state('loading')).toEqual(true);
		});

		it("should invoke 'fetch' with the correct params", () => {
			const mockUrl = 'https://langchat-crosspollination.herokuapp.com/api/v1/log_in/?format=json';
			const mockBody = {
				body: '{"username":"mockUsername","password":"mockPassword"}',
				headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
				method: 'POST'
			};
			instance.login();
			expect(fetch).toHaveBeenCalledWith(mockUrl, mockBody);
		});

		it('should throw an error if the response is not ok and save that error to redux store', async () => {
			window.fetch.mockImplementationOnce(() => Promise.reject(new Error()));
			await instance.login();
			expect(mockHandleError).toHaveBeenCalledWith("Your username or password is invalid. Please try again.");
		});

		it("should set the state of 'loading' back to false", async () => {
			wrapper.setState({ loading: true });
			expect(wrapper.state('loading')).toEqual(true);
			await instance.login();
			expect(wrapper.state('loading')).toEqual(false);
		});

		it("should invoke 'login' when login button is pressed", () => {
			const mockFn = jest.spyOn(instance, 'login');
			wrapper.find("[data-test='login-btn']").simulate('press');
			expect(mockFn).toHaveBeenCalled();
		});
	});
});

describe('mapDispatchToProps', () => {
	it('should call dispatch for changeUsername', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.changeUsername('mockFirstName');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.changeUsername('mockFirstName');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});

	it('should call dispatch for changePassword', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.changePassword('mockLastName');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.changePassword('mockLastName');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});

	it('should call dispatch for login', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.login('mockUser');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.login('mockUser');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});

	it('should call dispatch for handleError', () => {
		const mockDispatch = jest.fn();
		const actionToDispatch = actions.handleError('mock error message');
		const mappedProps = mapDispatchToProps(mockDispatch);
		mappedProps.handleError('mock error message');
		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
	});
});
