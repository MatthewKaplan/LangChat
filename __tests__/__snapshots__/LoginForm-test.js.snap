// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component should match snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "paddingTop": 50,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "600",
        "paddingBottom": 40,
        "paddingTop": 30,
      }
    }
  >
    Log In
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <Text />
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Username"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <Text />
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 80,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#007aff",
          "borderColor": "#007aff",
          "borderRadius": 25,
          "borderWidth": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
          "position": "absolute",
          "top": "84%",
          "width": "85%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontSize": 19,
            "fontWeight": "600",
            "paddingBottom": 12,
            "paddingTop": 12,
          }
        }
      >
        Log In
      </Text>
    </View>
  </View>
</View>
`;
